src/scenarios/scenario.manager.js

```diff
@@ -0,0 +1,41 @@
+import {fork} from 'child_process';
+import {stream as input_stream} from '../data-streams/input';
+import {stream as output_stream} from '../data-streams/output';
+
+const runningScenarios = new Map();
+
+input_stream.subscribe(notifyScenarios);
+
+function start(scenario) {
+    let scenarioProcess = fork(__dirname + '/runner');
+    runningScenarios.set(scenario.id, scenarioProcess);
+    scenarioProcess.send({
+        type: 'start',
+        content: scenario.body
+    });
+
+    scenarioProcess.on('message', ({type, content}) => {
+        if (type === 'message') {
+            output_stream.next(content);
+        }
+    });
+}
+
+function notifyScenarios(message) {
+    runningScenarios.forEach((scenario) => {
+        scenario.send({
+            type: 'message',
+            content: message
+        });
+    });
+}
+
+function stop(scenario) {
+    let scenarioProcess = runningScenarios.get(scenario.id);
+    if (scenarioProcess) {
+        scenarioProcess.kill();
+        runningScenarios.delete(scenario.id);
```

Comment:
scenario might have died on its own
like, due to some unhandled error
do we track such cases? if not -- lets **always** delete it from map in stop